contracts/ArrowVestingFactory.sol

```diff
@@ -12,7 +12,14 @@ contract ArrowVestingFactory {
 
     address immutable vestingImplementation;
```

Comment:
If `ArrowVestingBase` is upgradeable, can we really use `immutable` here? 

https://docs.openzeppelin.com/upgrades-plugins/1.x/faq#why-cant-i-use-immutable-variables

Reply:
I think this is fine, because `ArrowVestingFactory` is not an upgradable contract, so it can use immutable values without issue.

Reply:
Agreed @GoodluckH - should not be immutable